Sources/Purchasing/Purchases/PurchasesOrchestrator.swift

```diff
@@ -1481,3 +1489,26 @@ extension PurchasesOrchestrator {
     }
 
 }
+
+// MARK: - Application Lifecycle
+extension PurchasesOrchestrator {
+    func handleApplicationDidBecomeActive() {
+        if #available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, *),
+           self.observerMode && self.systemInfo.storeKitVersion == .storeKit2 {
+            Task(priority: .high) {
```

Comment:
So this will get notified every time the app becomes active, not only after becoming active after a purchase, do you think there could be any problem because we are checking for purchases everytime the app becomes active? I think it should be fine and it will actually check for purchases made outside the app in the Appstore not only via a regular purchase flow.

Reply:
I haven't been able to think of any reasons that it should cause any issues yet. The app shouldn't be backgrounded/foregrounded super quickly (like in a loop or something), so we won't be spamming SK2 with repeated requests, and each time this is called, we're only making at most 1 network request our backend, so I don't see an issue there either.

And yeah, I love that this will pick up transactions made outside the app! 😄

Reply:
not sure about task priority being high for observer mode, though, tbh. 
I wish there were better docs as to how this actually translates, but given that in observer mode you'd probably not be using our Entitlements in the first place, I kinda feel like `utility` might be a better fit?

Reply:
I mean, we can go with `high` and it's more likely than not that we'll be happy about it, I just want to ensure that we're being good citizens in people's apps

Reply:
I think that `utility` is appropriate given that observer mode usually isn't in most app's (if any) critical paths. I'll make the change!